main/mains.py

```diff
@@ -0,0 +1,83 @@
+import os
+import sys
+import inspect
+currentdir = os.path.dirname(os.path.abspath(
+    inspect.getfile(inspect.currentframe())))
+parentdir = os.path.dirname(currentdir)
+sys.path.insert(0, parentdir)
+
+from model.rooms import Office
+from model.rooms import LivingSpace
+from model.person import Fellow
+from model.person import Staff
+import random
+
+
+class Amity(object):
+    def __init__(self):
+        self.people = []
+        self.offices = []
+        self.living_spaces = []
+        self.unallocated = []
+        self.unallocated_fellows = []
+
+    def populate(self):
+        """prepopulates amity with 10 offices and 10 livingspaces"""
+        office_names = ['Valhalla', 'Oculus', 'Krypton', 'Shire',
+                        'Hogwarts', 'Mordor', 'Orange', 'Turquoise',
+                        'Peach', 'Cyan']
+
+        for i in range(0, len(office_names)):
+            self.offices.append(Office(office_names[i]))
+
+        living_names = ['Pearl', 'Ruby', 'Gem', 'Emerald', 'Sapphire',
+                        'Diamond', 'Graphite', 'Gold', 'Lithium', 'Chlorine']
+
+        for i in range(0, len(living_names)):
+            l = LivingSpace(living_names[i])
+            self.living_spaces.append(l)
+
+    def read_file(self, filename):
+        """Returns the people from the input file passed."""
+        input_file = open(filename)
+        for line in input_file:
+            # line is a line in the file filename
+            # slice the last character in line, as it is \n
+            line = line[:-1]
+            words = line.split(' ')
+            person_name = words[0] + ' ' + words[1]
+            if words[2] == 'FELLOW':
+                temp_choice = False
+                if words[3] == 'Y':
+                    temp_choice = True
+                fel = Fellow(person_name, temp_choice)
+                self.people.append(fel)
+            elif words[2] == 'STAFF':
+                self.people.append(Staff(person_name))
+            else:
+                raise Exception('Invalid input format')
+
+    def allocate_rooms(self):
+        """Allocates rooms"""
+        random.shuffle(self.people)
+        people_count = 0
+        people_len = len(self.people)
+        for office in self.offices:
+                while office.has_space():
+                    if people_count >= people_len:
+                        break
+                    office.add_person(self.people[people_count])
+                    people_count += 1
+
+                self.unallocated = self.people[people_count:]
```

Comment:
**Code Quality: Nitpick**

There is a problem here being that you are doing reassignments every time you run out of space in an office. You need to rewrite this logic to do a better job at ensuring we have no more rooms and there are still people to assign.